### PR TITLE
Fix user/application authentication docs

### PR DESCRIPTION
**Pull request by ryansch**
- Remove references to old param `auth_token` authentication.
- Add docs for deprecated header `auth_token` authentication.
- Add authentication section to User Endpoints page.

### CODE REVIEW

**Comment by ryansch** [comment]
@outstand/devs I can haz review?

File: user_endpoints.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # User Endpoints
+
+## Authentication
+
 The following endpoints are only available through OAuth2 Client Credentials Flow with an application with the `provision_users` scope.
 
+The endpoints described here are considered 'application level endpoints' which is why the oauth tokens you obtain through the client credentials flow aren't tied to a user.  These tokens _cannot_ be used to access 'user level endpoints' and regular user oauth tokens cannot be used to access these application level endpoints.
```
Comment (morganick) [inline user_endpoints.md]: The endpoints described here are considered 'application level endpoints' which is why the oauth tokens you obtain through the client credentials flow aren't tied to a user.  These tokens _cannot_ be used to access 'user level endpoints' (e.g contacts, groups, etc.) and regular user OAuth tokens cannot be used to access these application level endpoints.

**Comment by ryansch** [comment]
@morganick Ready for another look.

File: user_endpoints.md (hunk 1 above)
Comment (ryansch) [reply user_endpoints.md]: We chatted about this: users can use OAuth2.

**Comment by ryansch** [comment]
@morganick I can haz review?

**Pull request merged**